tiledb/libtiledb.pyx

```diff
@@ -1098,86 +1098,67 @@ def _tiledb_datetime_extent(begin, end):
 
 cdef bint _tiledb_type_is_datetime(tiledb_datatype_t tiledb_type) except? False:
     """Returns True if the tiledb type is a datetime type"""
-    if tiledb_type == TILEDB_DATETIME_YEAR or tiledb_type == TILEDB_DATETIME_MONTH \
-            or tiledb_type == TILEDB_DATETIME_WEEK or tiledb_type == TILEDB_DATETIME_DAY \
-            or tiledb_type == TILEDB_DATETIME_HR or tiledb_type == TILEDB_DATETIME_MIN \
-            or tiledb_type == TILEDB_DATETIME_SEC or tiledb_type == TILEDB_DATETIME_MS \
-            or tiledb_type == TILEDB_DATETIME_US or tiledb_type == TILEDB_DATETIME_NS \
-            or tiledb_type == TILEDB_DATETIME_PS or tiledb_type == TILEDB_DATETIME_FS \
-            or tiledb_type == TILEDB_DATETIME_AS:
-        return True
-    return False
-
+    return tiledb_type in (TILEDB_DATETIME_YEAR, TILEDB_DATETIME_MONTH, 
+        TILEDB_DATETIME_WEEK, TILEDB_DATETIME_DAY, TILEDB_DATETIME_HR, 
+        TILEDB_DATETIME_MIN, TILEDB_DATETIME_SEC, TILEDB_DATETIME_MS, 
+        TILEDB_DATETIME_US, TILEDB_DATETIME_NS, TILEDB_DATETIME_PS, 
+        TILEDB_DATETIME_FS, TILEDB_DATETIME_AS)
 
 def _tiledb_type_to_datetime(tiledb_datatype_t tiledb_type):
     """
     Return a datetime64 with appropriate unit for the given
     tiledb_datetype_t enum value
     """
-    if tiledb_type == TILEDB_DATETIME_YEAR:
-        return np.datetime64('', 'Y')
-    elif tiledb_type == TILEDB_DATETIME_MONTH:
-        return np.datetime64('', 'M')
-    elif tiledb_type == TILEDB_DATETIME_WEEK:
-        return np.datetime64('', 'W')
-    elif tiledb_type == TILEDB_DATETIME_DAY:
-        return np.datetime64('', 'D')
-    elif tiledb_type == TILEDB_DATETIME_HR:
-        return np.datetime64('', 'h')
-    elif tiledb_type == TILEDB_DATETIME_MIN:
-        return np.datetime64('', 'm')
-    elif tiledb_type == TILEDB_DATETIME_SEC:
-        return np.datetime64('', 's')
-    elif tiledb_type == TILEDB_DATETIME_MS:
-        return np.datetime64('', 'ms')
-    elif tiledb_type == TILEDB_DATETIME_US:
-        return np.datetime64('', 'us')
-    elif tiledb_type == TILEDB_DATETIME_NS:
-        return np.datetime64('', 'ns')
-    elif tiledb_type == TILEDB_DATETIME_PS:
-        return np.datetime64('', 'ps')
-    elif tiledb_type == TILEDB_DATETIME_FS:
-        return np.datetime64('', 'fs')
-    elif tiledb_type == TILEDB_DATETIME_AS:
-        return np.datetime64('', 'as')
-    else:
+    tiledb_type_to_datetime = {
```

Comment:
Minor nit: this should be defined out of line, because otherwise the dictionary is constructed on every function call (I don't think Cython does any static optimization here, but I could be wrong - CPython does not).